Marlin/src/MarlinCore.cpp

```diff
@@ -937,6 +937,14 @@ void minkill(const bool steppers_off/*=false*/) {
     // Reboot the board
     hal.reboot();
 
+  #elif defined(PLR_REBOOT_TIMEOUT)
+
+    for (uint16_t i = 0; i < PLR_REBOOT_TIMEOUT; i++) {
+      hal.watchdog_refresh();
+      delay(1000);
+    }
+    hal.reboot();
+
```

Comment:
The way this is implemented any and all KILL events will cause the machine to reboot here. It is not limited to a power loss situation. There should probably be a flag set to indicate that a power loss (with backup power) has occurred.

Reply:
Ok, I'll take a look, thanks.

Reply:
I would add flag to `PrintJobRecovery` class. like 

```
class PrintJobRecovery {
  public:
static bool outage;

```
Sounds reasonable?